irc/server.go

```diff
@@ -226,9 +229,105 @@ func (server *Server) checkTorLimits() (banned bool, message string) {
 
 // createListener starts a given listener.
 func (server *Server) createListener(addr string, conf listenerConfig, bindMode os.FileMode) (*ListenerWrapper, error) {
-	// make listener
+	if conf.WebSocket {
+		return server.createWSListener(addr, conf)
+	}
+	return server.createNetListener(addr, conf, bindMode)
+}
+
+func (server *Server) isTrusted(ip string) bool {
+	netIP := net.ParseIP(ip)
```

Comment:
`RemoteAddr` sucks. It has "no defined format". You can't parse it directly with `ParseIP`, you need `net.SplitHostPort` and then a fallback for Unix domain sockets (it will generally be `@` in that case --- you can fall back to interpreting it as 127.0.0.1, see `AddrToIP` in the oragono codebase). See this PR:

https://github.com/kiwiirc/webircgateway/pull/65